### PR TITLE
Fix `pandas` and plotting warning

### LINKED ISSUE #958: Matplotlib's `draw_all` is deprecated
## Description

The use of `draw_all` [here](https://github.com/theislab/scvelo/blob/ce52f5846e6e8f5953ebd1e02ff675ef9bdc5dd3/scvelo/plotting/utils.py#L869) will be deprecated:

`MatplotlibDeprecationWarning: The draw_all function was deprecated in Matplotlib 3.6 and will be removed two minor releases later. Use fig.draw_without_rendering() instead.`

### LINKED ISSUE #1037: AttributeError in scv.pl.velocity_embedding_stream
plotting raises Attribute error

```python
scv.pl.velocity_embedding_stream(adata, basis='umap')
```

<details>

<summary> Error output </summary>

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[9], line 1
----> 1 scv.pl.velocity_embedding_stream(adata, basis='umap')

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/scvelo/plotting/velocity_embedding_stream.py:252, in velocity_embedding_stream(adata, basis, vkey, density, smooth, min_mass, cutoff_perc, arrow_color, arrow_size, arrow_style, max_length, integration_direction, linewidth, n_neighbors, recompute, color, use_raw, layer, color_map, colorbar, palette, size, alpha, perc, X, V, X_grid, V_grid, sort_order, groups, components, legend_loc, legend_fontsize, legend_fontweight, xlabel, ylabel, title, fontsize, figsize, dpi, frameon, show, save, ax, ncols, **kwargs)
    249 ax.streamplot(X_grid[0], X_grid[1], V_grid[0], V_grid[1], **stream_kwargs)
    251 size = 8 * default_size(adata) if size is None else size
--> 252 ax = scatter(
    253     adata,
    254     layer=layer,
    255     color=color,
    256     size=size,
    257     title=title,
    258     ax=ax,
    259     zorder=0,
    260     **scatter_kwargs,
    261 )
    262 if save is True:
    263     save = "embedding_stream"

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/scvelo/plotting/scatter.py:668, in scatter(adata, basis, x, y, vkey, color, use_raw, layer, color_map, colorbar, palette, size, alpha, linewidth, linecolor, perc, groups, sort_order, components, projection, legend_loc, legend_loc_lines, legend_fontsize, legend_fontweight, legend_fontoutline, legend_align_text, xlabel, ylabel, title, fontsize, figsize, xlim, ylim, add_density, add_assignments, add_linfit, add_polyfit, add_rug, add_text, add_text_pos, add_margin, add_outline, outline_width, outline_color, n_convolve, smooth, normalize_data, rescale_color, color_gradients, dpi, frameon, zorder, ncols, nrows, wspace, hspace, show, save, ax, **kwargs)
    666     if not (add_outline is None or g_bool is None):
    667         groups = add_outline
--> 668     set_legend(
    669         adata,
    670         ax,
    671         color,
    672         legend_loc,
    673         scatter_array,
    674         legend_fontweight,
    675         legend_fontsize,
    676         legend_fontoutline,
    677         legend_align_text,
    678         groups,
    679     )
    680 if add_density:
    681     plot_density(x, y, add_density, ax=ax)

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/scvelo/plotting/utils.py:555, in set_legend(adata, ax, value_to_plot, legend_loc, scatter_array, legend_fontweight, legend_fontsize, legend_fontoutline, legend_align_text, groups)
    553     legend_fontoutline = 1
    554 obs_vals = adata.obs[value_to_plot]
--> 555 obs_vals.cat.categories = obs_vals.cat.categories.astype(str)
    556 color_keys = adata.uns[f"{value_to_plot}_colors"]
    557 if isinstance(color_keys, dict):

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/pandas/core/base.py:178, in NoNewAttributesMixin.__setattr__(self, key, value)
    172 if getattr(self, "__frozen", False) and not (
    173     key == "_cache"
    174     or key in type(self).__dict__
    175     or getattr(self, key, None) is not None
    176 ):
    177     raise AttributeError(f"You cannot add any new attribute '{key}'")
--> 178 object.__setattr__(self, key, value)

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/pandas/core/accessor.py:99, in PandasDelegate._add_delegate_accessors.<locals>._create_delegator_property.<locals>._setter(self, new_values)
     98 def _setter(self, new_values):
---> 99     return self._delegate_property_set(name, new_values)

File ~/miniconda3/envs/scVelo/lib/python3.10/site-packages/pandas/core/arrays/categorical.py:2460, in CategoricalAccessor._delegate_property_set(self, name, new_values)
   2459 def _delegate_property_set(self, name, new_values):
-> 2460     return setattr(self._parent, name, new_values)

AttributeError: can't set attribute 'categories'
```

</details>

<details> <summary> Versions </summary>

```pytb
scvelo==0.2.5  scanpy==1.9.3  anndata==0.9.1  loompy==3.0.6  numpy==1.23.5  scipy==1.10.1  matplotlib==3.7.1  sklearn==1.2.2  pandas==2.0.0 
```

</details>

### LINKED ISSUE #1058: scv.tl.recover_dynamics fails
When trying to run `scv.tl.recover_dynamics(adata, n_jobs=8)`, I get 

```pytb
    761 if dm is not None and dm.recoverable:
    762     dm.m = m[idx]
    763     dm.alpha, dm.beta, dm.gamma, dm.pars[:3] = (
--> 764         np.array([dm.alpha, dm.beta, dm.gamma, dm.pars[:3]]) / dm.m[-1]
    765     )
    766     dm.t, dm.tau, dm.t_, dm.pars[4] = (
    767         np.array([dm.t, dm.tau, dm.t_, dm.pars[4]]) * dm.m[-1]
    768     )
    770 return adata if copy else dm

ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (4,) + inhomogeneous part.
```
This appears to be a similar error to https://github.com/theislab/scvelo/issues/1044

<details>

<summary> Error output </summary>

```pytb
# paste the error output here, if applicable
```

</details>

<details> <summary> Versions </summary>

```pytb
scvelo==0.3.0  scanpy==1.9.3  anndata==0.8.0  loompy==3.0.7  numpy==1.24.2  scipy==1.10.1  matplotlib==3.7.0  sklearn==1.1.3  pandas==1.5.3  
```

</details>

### LINKED ISSUE #1071: Unit tests in `test_basic.py` failing
The unit tests `test_pipeline` and `test_highly_variable_subset` are currently failing. See _e.g._, [here](https://github.com/theislab/scvelo/actions/runs/5137575253/jobs/9245873966).

### PR DESCRIPTION
**Pull request by michalk8**
## Bug fixes

Fixes `pandas>=2`, removes setting attributes in `get_df`, `Colorbar.draw()` (deprecated; doesn't seem to have any effect) and uses `scanpy.pp.log1p` in `log1p` implementation. Also fixes some tests and ignores warning in tests.

Also incorporates the changes from #1059 

## Related issues

Closes #958
Closes #1037
Closes #1058 
Closes https://github.com/theislab/cellrank/issues/1018
Closes #1071

### CODE REVIEW

File: tests/preprocessing/test_neighbors.py
Comment (WeilerP) [inline tests/preprocessing/test_neighbors.py:1]: This part is also unrelated to the Pandas and plotting issue, right? Could you please suggest them in a separate PR (e.g., together with the changes to the moments tests)?

File: scvelo/preprocessing/utils.py [hunk 1]
```diff
@@ -587,14 +585,10 @@ def log1p(data, copy=False):
...
-        (adata.X.data if issparse(adata.X) else adata.X)
-        if isinstance(adata, AnnData)
-        else adata
-    )
-    np.log1p(X, out=X)
-    return adata if copy else None
+    from scanpy.pp import log1p as _log1p
+
+    res = _log1p(data, copy=copy)
+    return res if copy else None
```
[the remark this answers is not in the document]
Comment (WeilerP) [reply scvelo/preprocessing/utils.py:591]: I was meaning to remove `log1p` all together from scvelo. Unless these changes are required to fix the Pandas and plotting warnings, can you please revert the changes and open an issue, instead?

File: tests/preprocessing/test_utils.py
Comment (WeilerP) [inline tests/preprocessing/test_utils.py:1]: If the changes to `log1p` are reverted, this part can be undone as well.

**Review by WeilerP — COMMENTED** [review]
Thanks, @michalk8. I only left minor comments (mainly moving unnecessary changes to separate PRs).

**Comment by hiraksarkar** [comment]
Getting the same error with the bellow version 

```
>  pd.__version__
'2.0.3'
> scv.__version__
'0.2.5'
```
